sound/soc/sof/intel/cnl.c

```diff
@@ -83,7 +83,8 @@ static irqreturn_t cnl_ipc_irq_thread(int irq, void *context)
 	}
 
 	/* new message from DSP */
-	if (hipctdr & CNL_DSP_REG_HIPCTDR_BUSY) {
+	if (hipctdr & CNL_DSP_REG_HIPCTDR_BUSY &&
+	    hipcctl & CNL_DSP_REG_HIPCCTL_BUSY) {
```

Comment:
no, we've been there before. The hipcctl register controls the generation of the interrupt, so it's useless to test in the interrupt if the interrupt is enabled. NAK on this one.

Reply:
let me remove this commit after some experiments here, as written in the commit message, this may be not needed.